### PR TITLE
Add mutable slice Read()

### PR DESCRIPTION
**Pull request by kaidokert**
Adds a Read() impl for mutable u8 slices as well. This allows passing slices around for functions with both <T: Read + Write> constraints

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -498,6 +498,15 @@ impl<'a> Read for &'a [u8] {
     }
 }
 
+impl<'a> Read for &'a mut [u8] {
+    type ReadError = Void;
+
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize, Self::ReadError> {
+        let mut immutable : &[u8] = self;
+        immutable.read(buf)
+    }
```
Comment (Kixunil) [inline src/lib.rs:507]: This doesn't update `self` to point to unread part afterwards.

**Review by Kixunil — CHANGES_REQUESTED** [review]
Makes sense! Unfortunately your code has a bug.

File: src/lib.rs (hunk 1 above)
Comment (kaidokert) [reply src/lib.rs:511]: Indeed, thank for pointing that out. I was really stumped as to fix this properly, came up with `core::mem::replace` trick - please have a look if that looks reasonable.

Comment (Kixunil) [reply src/lib.rs:511]: Yep, that's the usual way to solve this.

**Review by Kixunil — APPROVED** [review]
Looks good, thanks!

**Pull request merged**